### PR TITLE
Reflect `SmolStr`'s `De/Serialize` implementation

### LINKED ISSUE #14969: SmolStr doesn't have ReflectDeserialize
## What problem does this solve or what need does it fill?

- Deserializing structs via reflection that have `SmolStr` in them.

## What solution would you like?

Add `ReflectDeserialize` to `SmolStr`.

## What alternative(s) have you considered?

Workaround from `Giooschi` on discord: "As a workaround for now you can call `app.register_type_data::<SmolStr, ReflectDeserialize>()` or the equivalent method on `TypeRegistry`"

## Additional context

The error I encountered was: `"the TypeRegistration for smol_str::SmolStr doesn't have ReflectDeserialize"`.

### PR DESCRIPTION
**Pull request by bushrat011899**
# Objective

- Fixes #14969

## Solution

- Added `Deserialize` to the list of reflected traits for `SmolStr`

## Testing

- CI passed locally.

### CODE REVIEW

**Review by killercup — APPROVED** [review]
LGTM. Serde is already in tree.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
`Serialize` should be reflected here too.

File: crates/bevy_reflect/src/impls/smol_str.rs [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+use crate::{std_traits::ReflectDefault, ReflectDeserialize};
 use bevy_reflect_derive::impl_reflect_value;
 
-impl_reflect_value!(::smol_str::SmolStr(Debug, Hash, PartialEq, Default));
+impl_reflect_value!(::smol_str::SmolStr(
+    Debug,
+    Hash,
+    PartialEq,
+    Default,
+    Deserialize,
```
Comment (alice-i-cecile) [inline crates/bevy_reflect/src/impls/smol_str.rs:10]:
```suggestion
    Serialize,
    Deserialize,
```

Comment (bushrat011899) [reply crates/bevy_reflect/src/impls/smol_str.rs:11]: Oh yeah that's perfectly reasonable, don't know how I missed that!

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**